akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/scaladsl/NewEntityWithClusterShardingSpec.scala

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.scaladsl
+
+import scala.concurrent.duration._
+import scala.util.Failure
+import scala.util.Success
+
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.testkit.typed.scaladsl.TestProbe
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Entity
+import akka.actor.typed.EntityEnvelope
+import akka.actor.typed.EntityMessageExtractor
+import akka.actor.typed.EntityTypeKey
+import akka.actor.typed.PostStop
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter._
+import akka.cluster.MemberStatus
+import akka.cluster.typed.Cluster
+import akka.cluster.typed.Join
+import akka.cluster.typed.Leave
+import akka.pattern.AskTimeoutException
+import akka.serialization.jackson.CborSerializable
+import akka.util.Timeout
+import akka.util.ccompat._
+import com.typesafe.config.ConfigFactory
+import org.scalatest.wordspec.AnyWordSpecLike
+
+@ccompatUsedUntil213
+object NewEntityWithClusterShardingSpec {
```

Comment:
This is a copy of ClusterShardingSpec, but then using the new types. All tests are passing!

We may want to remove this and refactor ClusterShardingSpec to use new API.